src/User/Repository/UserRepository.php

```diff
@@ -136,4 +136,36 @@ public function getUsers(): array
             throw new  DatabaseException(sprintf('Error while fetching users: %s', $e->getMessage()));
         }
     }
+
+    /**
+     * @return UserInterface[]
+     *
+     * @throws DatabaseException
+     */
+    public function searchUser(string $searchQuery): array
+    {
+        $list = [];
+        $q = '%' . $searchQuery . '%';
+
+        try {
+            $userListing = new UserListing();
+            $userListing->setCondition(
+                'name LIKE ? OR firstname LIKE ? OR lastname LIKE ? OR email LIKE ? or id = ?',
+                [$q, $q, $q, $q, (int)$searchQuery]
```

Comment:
Hmm, is it really necessary that you could search for the ID? 
Like then you can basically found out every id of every user no just by "brute forcing" the search?